src/components/AddTodoForm/AddTodoForm.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react';
+import usersFromServer from './../../api/users';
+import { Todo } from '../../types/Todo';
+import { getUserById } from '../../utils/getUserById';
+import { getNewTodoId } from '../../utils/getNewTodoId';
+
+interface Props {
+  onAdd: (todo: Todo) => void;
+  todos: Todo[];
+}
+
+enum TypeErrors {
+  Default = '',
+  TitleError = 'Please enter a title',
+  UserIdError = 'Please choose a user',
+}
+
+export const AddTodoForm: React.FC<Props> = ({ onAdd, todos }) => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(TypeErrors.Default);
+
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(TypeErrors.Default);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(TypeErrors.Default);
+  };
+
+  const handleChangeUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setUserIdError(TypeErrors.Default);
+  };
+
+  const handleReset = () => {
+    setTitle('');
+    setTitleError(TypeErrors.Default);
+
+    setSelectedUserId(0);
+    setUserIdError(TypeErrors.Default);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError(TypeErrors.TitleError);
+    }
+
+    if (!selectedUserId) {
+      setUserIdError(TypeErrors.UserIdError);
+    }
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getNewTodoId(todos),
+      title: title,
+      completed: false,
+      userId: selectedUserId,
+      user: getUserById(selectedUserId),
```

Comment:
[REACT KNOWLEDGE] - Don't render the component if the property that you pass to the component has `null` or `undefined` value. Since `getUserById` might return `undefined` if the user is not found, you should handle this case appropriately and avoid passing `undefined` to the `user` property of the `Todo` object.